site/src/sphinx/advanced-retry.rst

```diff
@@ -1,8 +1,228 @@
+                }
+                return Optional.empty(); // Return no backoff not to retry anymore
+            });
+        }
+    };
+
+This will retry when the response's status is ``409`` or ResponseTimeoutException_ is raised.
+
+.. note::
+
+    We declare a Backoff_ as a member and reuse it when a ``strategy`` returns it, so that we do not return
+    a different Backoff_ instance for each ``shouldRetry()``. RetryingClient_ internally tracks the
+    reference of the returned Backoff_ and increases the counter that keeps the number of attempts made so far,
+    and resets it to 0 when the Backoff_ returned by the strategy is not same as before. Therefore, it is
+    important to return the same Backoff_ instance unless you decided to change your Backoff_ strategy, which
+    yields a different delay based on the number of retried, will not work as expected. We will take
+    a close look what the Backoff_ is at the bottom of this page.
+
+You can return different Backoff_ according to the response.
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.ResponseTimeoutException;
+    import com.linecorp.armeria.common.HttpStatusClass;
+
+    new RetryStrategy<HttpRequest, HttpResponse>() {
+        final Backoff backoffOnServerErrorOrTimeout = RetryStrategy.defaultBackoff;
+        final Backoff backoffOnConflict = Backoff.fixed(100);
+
+        @Override
+        public CompletableFuture<Optional<Backoff>> shouldRetry(HttpRequest request,
+                                                                HttpResponse response) {
+            return response.aggregate().handle((result, cause) -> {
+                if (cause != null) {
+                    if (cause instanceof ResponseTimeoutException) {
+                        return Optional.of(backoffOnServerErrorOrTimeout);
+                    }
+                } else if (result.headers().status().codeClass() == HttpStatusClass.SERVER_ERROR) {
+                    return Optional.of(backoffOnServerErrorOrTimeout);
+                } else if (result.headers().status() == HttpStatus.CONFLICT) {
+                    return Optional.of(backoffOnConflict);
+                }
+                return Optional.empty();
+            });
+        }
+    };
+
+
+Retry on ResponseTimeoutException_
+----------------------------------
+
+ResponseTimeoutException_ can occur in two different situations while retrying. First, it can occur when the
+time of whole retry session has passed the time previously configured using:
+
+.. code-block:: java
+
+    ClientBuilder.defaultResponseTimeoutMillis(millis);
+
+You cannot retry on this ResponseTimeoutException_.
+Second, it can occur when the time of individual attempt in retry has passed the time configured when
+creaing the decorator with:
+
+.. code-block:: java
+
+    RetryingHttpClient.newDecorator(strategy, defaultMaxAttempts, responseTimeoutMillisForEachAttempt);
+
+You can retry on this ResponseTimeoutException_.
+
```

Comment:
How about adding a simple example diagram here? e.g. when making a retrying request to an unresponsive service with responseTimeoutMillis = 10,000 and responseTimeoutMillisForEachAttempt = 3,000

```
0ms         3,000ms     6,000ms     9,000ms
+-----------+-----------+-----------+---+
| Attempt 1 | Attempt 2 | Attempt 3 | A |
+-----------+-----------+-----------+---+
                                        10,000ms (4th attempt is aborted and you get ResponseTimeoutException)

```